src/openloco/windows/IndustryWindow.cpp

```diff
@@ -0,0 +1,1064 @@
+                {
+                    if (yPos1 < self->y + self->height - 7)
+                    {
+                        if (yPos2 < self->y + self->height - 7)
+                        {
+                            gfx::draw_line(dpi, xPos, yPos1, xPos + 1, yPos2, colour::get_shade(self->colours[1], 7));
+                        }
+                    }
+                }
+                month--;
+                if (month < 0)
+                {
+                    month = 11;
+                    year--;
+
+                    yearSkip++;
+                    if (yearSkip >= 3)
+                        yearSkip = 0;
+                }
+            }
+        }
+
+        // 0x00456505
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameIndustryPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_industry:
+                case common::widx::tab_production:
+                case common::widx::tab_production_2:
+                case common::widx::tab_transported:
+                    common::switchTab(self, widgetIndex);
+                    break;
+            }
+        }
+
+        // 0x0045654F
+        static void on_resize(window* self)
+        {
+            {
+                self->set_size(minWindowSize, maxWindowSize);
+            }
+        }
+
+        static void initEvents()
+        {
+            events.draw = draw;
+            events.on_mouse_up = on_mouse_up;
+            events.on_resize = on_resize;
+            events.on_update = common::update;
+            events.prepare_draw = prepare_draw;
+            events.text_input = common::text_input;
+        }
+    }
+
+    namespace production_2
+    {
+        static const gfx::ui_size_t minWindowSize = { 299, 282 };
+
+        static const gfx::ui_size_t maxWindowSize = { 299, 337 };
+
+        static widget_t widgets[] = {
+            commonWidgets(222, 136, string_ids::title_industry_monthly_production),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        // 0x0045626F
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+
+            common::repositionTabs(self);
+        }
+
+        // 0x0045630F
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            // Draw Units of Cargo sub title
+            auto industry = industrymgr::get(self->number);
+            auto industryObj = objectmgr::get<industry_object>(industry->object_id);
+            auto cargoObj = objectmgr::get<cargo_object>((uint8_t)industryObj->produced_cargo[1]);
```

Comment:
See above comments. Also is there a way we can make a common function for these two tabs.